www/class/centreonCustomView.class.php

```diff
@@ -442,9 +442,26 @@ public function deleteCustomView($viewId)
                 }
                 //if owner not delete
             } else {
-                $query = 'UPDATE custom_view_user_relation SET is_consumed = 0 ' .
-                    'WHERE custom_view_id = :viewId AND user_id = :userId ';
-                $stmt = $this->db->prepare($query);
+                // delete widget pref
+                $stmt = $this->db->prepare(
+                    'DELETE FROM widget_preferences 
+                    WHERE user_id = :userId 
+                    AND widget_view_id IN (
+                        SELECT widget_view_id FROM widget_views WHERE custom_view_id = :viewId
+                    )'
+                );
+                $stmt->bindParam(':userId', $this->userId, \PDO::PARAM_INT);
+                $stmt->bindParam(':viewId', $viewId, \PDO::PARAM_INT);
+                $dbResult = $stmt->execute();
+                if (!$dbResult) {
```

Comment:
PDO is configured with PDO::ERRMODE_EXCEPTION flag, then it will throw an exception if query fails.
Need to wrap it in try .. catch

Reply:
seems not resolved